annotated_test.go

```diff
@@ -1440,12 +1440,62 @@ func TestAnnotate(t *testing.T) {
 					fx.ResultTags(`name:"AA"`),
 				),
 			),
-			fx.Invoke(newB),
+			fx.Invoke(
+				fx.Annotate(func(a, aa *a) (*b, *b) {
+					return newB(a), newB(aa)
+				}, fx.ParamTags(`name:"A"`, `name:"AA"`))),
 		)
 
 		err := app.Err()
-		require.Error(t, err)
-		assert.Contains(t, err.Error(), "encountered error while applying annotation using fx.Annotate to go.uber.org/fx_test.TestAnnotate.func1(): cannot apply more than one line of ResultTags")
+		require.NoError(t, err)
+		defer app.RequireStart().RequireStop()
+	})
+
+	t.Run("specify two ResultTags containing multiple tags", func(t *testing.T) {
```

Comment:
We should add a test case for using multiple As annotations and ResultTags annotations. For example, something like the following:
```go
fx.Provide(
    fx.Annotate(
        func() *bytes.Buffer { return bytes.NewBuffer() },
        fx.As(new(io.Reader)),
        fx.As(new(io.Writer)),
        fx.ResultTags(`name:"foo"`),
        fx.ResultTags(`name:"bar"`),
    )
)
```
In fact, what's the expected behavior in this case?
I think the current implementation will produce 4 results from the above code, right? But does that make sense? Does "using multiple ResultTags annotation" only make sense when there is 0 or 1 As annotation? If so, we should probably add additional logic to validate that to apply() method of resultTagsAnnotation.